src/journal/journalctl.c

```diff
@@ -2099,44 +2166,13 @@ static int show(Context *c) {
                                 break;
                 }
 
-                if (arg_until_set && !arg_reverse && (arg_lines < 0 || arg_since_set)) {
```

Comment:
As mentioned in https://github.com/systemd/systemd/issues/31776#issuecomment-1997010061, relax this check should fix the problem? Maybe, replace since_seeked with a generic `seek_need_recheck` field, that's set to true if cursor is used or --until is used with --lines. In all other cases it's guaranteed that the initial seek is at right place.

Reply:
Maybe, but not sure. The condition has been extended, and is now already complicated. I think it is better to simplify the logic, rather than extend it further.

Reply:
It actually gave me a hard time trying to understand the implications :/